query/search.js

```diff
@@ -119,4 +60,67 @@ function generate( params ){
   return query;
 }
 
+function addParsedMatch(query, defaultInput, parsedInput) {
+  query.query.filtered.query.bool.should = [];
+
+  // copy expected admin fields so we can remove them as we parse the address
+  var unmatchedAdminFields = adminFields.slice();
+
+  // address
+  // number, street, postalcode
+  addMatch(query, unmatchedAdminFields, 'address.number', parsedInput.number, addressWeights.number);
+  addMatch(query, unmatchedAdminFields, 'address.street', parsedInput.street, addressWeights.street);
+  addMatch(query, unmatchedAdminFields, 'address.zip', parsedInput.postalcode, addressWeights.zip);
+
+  // city
+  // admin2, locality, local_admin, neighborhood
+  addMatch(query, unmatchedAdminFields, 'admin2', parsedInput.admin2, addressWeights.admin2);
```

Comment:
please check if its ```parsedInput.city``` or ```parsedInput.admin2``` - I remember ```addressIt``` has no notion of admins.. 

I just looked at master - [this](https://github.com/pelias/api/blob/master/query/search.js#L69) may be a bug in master because ```parsedInput.admin2``` would always be ```undefined```